libs/vm-modules/src/ml/model/model.cpp

```diff
@@ -395,29 +417,34 @@ void VMModel::AssertLayerTypeMatches(SupportedLayerType                layer,
   };
   if (std::find(valids.begin(), valids.end(), layer) == valids.end())
   {
-    std::string const message{"Invalid params specified for \"" + LAYER_NAMES_.at(layer) +
-                              "\" layer."};
-    vm_->RuntimeError(message);
-    throw std::runtime_error(message);
+    throw std::runtime_error("Invalid params specified for \"" + LAYER_NAMES_.at(layer) +
+                             "\" layer.");
   }
 }
 
 VMModel::SequentialModelPtr VMModel::GetMeAsSequentialIfPossible()
 {
   if (model_category_ != ModelCategory::SEQUENTIAL)
   {
-    std::string const message{"No \"add\" method exists for non-sequential models!"};
-    vm_->RuntimeError(message);
-    throw std::runtime_error(message);
+    throw std::runtime_error("Layer adding is allowed only for sequential models!");
```

Comment:
same as above

Reply:
This function is also always covered by catch blocks, either AddLayer<> or Compile... one, so the error message will be shown neatly.